sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateScanBuilder.scala

```diff
@@ -67,8 +67,7 @@ class StateScan(
     })
 
     if (partitions.headOption.isEmpty) {
-      throw new IllegalArgumentException("The state does not have any partition. Please double " +
-        s"check that the query points to the valid state. options: $sourceOptions")
+      throw StateDataSourceErrors.noPartitionDiscoveredInStateStore(sourceOptions)
```

Comment:
Can't you reuse `SparkException.internalError()`?

Reply:
How we deal with internal error for data source? Do we think it won't matter if we use a single error class for all of internal errors, among engine and data sources?